core/src/main/java/org/mule/runtime/core/internal/streaming/ManagedCursorProvider.java

```diff
@@ -55,20 +84,25 @@ public final T openCursor() {
   protected abstract T managedCursor(T cursor, CursorContext handle);
 
   @Override
-  public void releaseResources() {
-    delegate.releaseResources();
+  public final void releaseResources() {
+    janitor.releaseResources();
+  }
+
+  public CursorProviderJanitor getJanitor() {
+    return janitor;
   }
 
   @Override
   public void close() {
-    delegate.close();
+    janitor.close();
   }
 
   @Override
   public boolean isClosed() {
     return delegate.isClosed();
   }
 
+  //TODO: Seems no longer necessary
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7643&issues=AWirZZm9otIXnmsUDp_T&open=AWirZZm9otIXnmsUDp_T)

[ISSUE_KEY]: AWirZZm9otIXnmsUDp_T